include/onnxruntime/core/providers/nnapi/nnapi_provider_factory.h

```diff
@@ -1,15 +1,35 @@
 // Copyright 2019 JD.com Inc. JD AI
+#pragma once
 
 #include "onnxruntime_c_api.h"
 
+// NNAPIFlag are bool options we want to set for NNAPI EP
+// To generate a unsigned long nnapi_flags for using with OrtSessionOptionsAppendExecutionProvider_Nnapi below,
+// If using C
+//   unsigned long nnapi_flags = 0;
+//   nnapi_flags |= 1UL << NNAPI_FLAG_USE_FP16;
+// If using C++
+//   std::bitset<NNAPI_FLAG_MAX> _nnapi_flags;
+//   _nnapi_flags.set(NNAPI_FLAG_USE_FP16);
+//   unsigned long nnapi_flags = _nnapi_flags.to_ulong();
+enum NNAPIFlag {
```

Comment:
nit: Prefer the enum values do the bitshifting instead of making the user do it. Latter seems potentially more error prone than just telling them to '|' the values they want to set.

e.g.
  NNAPI_FLAG_NONE = 0,
  NNAPI_FLAG_USE_FP16 = 1 << 0,
  NNAPI_FLAG_USE_NCHW = 1 << 1,
  ... = 1 << 2, etc. etc.

Also should the enum name be 'NNAPIFlags'?

Reply:
Move to use bit flags
Removed bitset since this won't be easily used with bit flags